ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/bridge/ConflictCheckingLedgerBridge.scala

```diff
@@ -0,0 +1,362 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml
+package ledger.sandbox.bridge
+
+import akka.NotUsed
+import akka.stream.scaladsl.Flow
+import com.daml.error.{ContextualizedErrorLogger, DamlContextualizedErrorLogger}
+import com.daml.ledger.offset.Offset
+import com.daml.ledger.participant.state.index.v2.IndexService
+import com.daml.ledger.participant.state.v2.Update.CommandRejected.FinalReason
+import com.daml.ledger.participant.state.v2.{CompletionInfo, Update}
+import com.daml.ledger.sandbox.bridge.ConflictCheckingLedgerBridge._
+import com.daml.ledger.sandbox.bridge.LedgerBridge.{fromOffset, successMapper, toOffset}
+import com.daml.ledger.sandbox.bridge.SequencerState.LastUpdatedAt
+import com.daml.ledger.sandbox.domain.Rejection._
+import com.daml.ledger.sandbox.domain.Submission.Transaction
+import com.daml.ledger.sandbox.domain._
+import com.daml.lf.data.Ref
+import com.daml.lf.data.Time.Timestamp
+import com.daml.lf.engine.Blinding
+import com.daml.lf.transaction.{Transaction => LfTransaction}
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.metrics.Timed
+import com.daml.platform.ApiOffset
+import com.daml.platform.apiserver.execution.MissingContracts
+import com.daml.platform.server.api.validation.ErrorFactories
+import com.daml.platform.store.appendonlydao.events._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.chaining._
+import scala.util.{Failure, Success, Try}
+
+private[sandbox] class ConflictCheckingLedgerBridge(
+    participantId: Ref.ParticipantId,
+    indexService: IndexService,
```

Comment:
I suspect of using the IndexService for anything at this point as it is downstream from the ledger bridge (should we not be using the sequence state) ?

Reply:
I now see the design uses a blend of the in-flight sequenced state which overrides the data in the ledger.  The in-flight state being pruned once the offset has been indexed.